src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Extract.cs

```diff
@@ -209,5 +216,2406 @@ public async Task DirectoryEntryWithData(bool async)
+        //        {
+        //            using var _ = e.Open(archivePassword);
+        //        });
+        //    }
+        //}
+
+        [Fact]
+        public void OpenAESEncryptedTxtFile_ShouldReturnPlaintextWinZip()
+        {
+            string zipPath = Path.Join(DownloadsDir, "source_plain_winzip.zip");
+            using var archive = ZipFile.OpenRead(zipPath);
+
+            var entry = archive.Entries.First(e => e.FullName.EndsWith("source_plain.txt"));
+            using var stream = entry.Open("123456789");
+            using var reader = new StreamReader(stream);
+            string content = reader.ReadToEnd();
+
+            Assert.Equal("this is plain", content);
+        }
+
+        [Fact]
+        public void OpenAESEncryptedTxtFile_ShouldReturnPlaintextWinRar()
+        {
+            string zipPath = Path.Join(DownloadsDir, "source_plain.zip");
+            using var archive = ZipFile.OpenRead(zipPath);
+
+            var entry = archive.Entries.First(e => e.FullName.EndsWith("source_plain.txt"));
+            using var stream = entry.Open("123456789");
+            using var reader = new StreamReader(stream);
+            string content = reader.ReadToEnd();
+
+            Assert.Equal("this is plain", content);
+        }
+
+        [Fact]
+        public void OpenAESEncryptedTxtFile_ShouldReturnPlaintext7zip()
+        {
+            string zipPath = Path.Join(DownloadsDir, "source_plain7z.zip");
+            using var archive = ZipFile.OpenRead(zipPath);
+
+            var entry = archive.Entries.First(e => e.FullName.EndsWith("source_plain.txt"));
+            using var stream = entry.Open("123456789");
+            using var reader = new StreamReader(stream);
+            string content = reader.ReadToEnd();
+
+            Assert.Equal("this is plain", content);
+        }
+
+        [Fact]
+        public void OpenMultipleAESEncryptedEntries_ShouldReturnCorrectContent()
+        {
+            string zipPath = Path.Join(DownloadsDir, "multiple_entries_aes.zip");
+            using var archive = ZipFile.OpenRead(zipPath);
+
+            // Test first entry
+            var entry1 = archive.Entries.First(e => e.FullName.EndsWith("source_plain.txt"));
+            using (var stream1 = entry1.Open("123456789"))
+            using (var reader1 = new StreamReader(stream1))
+            {
+                string content1 = reader1.ReadToEnd();
+                Assert.Equal("this is plain", content1);
+            }
+
+            // Test second entry
+            var entry2 = archive.Entries.First(e => e.FullName.EndsWith("source_plain_2.txt"));
+            using (var stream2 = entry2.Open("123456789"))
+            using (var reader2 = new StreamReader(stream2))
+            {
+                string content2 = reader2.ReadToEnd();
+                Assert.Equal("this is plain", content2);
+            }
+        }
+
+        [Fact]
+        public async Task CreateAndReadAES256EncryptedEntry_RoundTrip()
```

Comment:
You can parameterize tests on the encryption method to remove test duplication.

In this case, since we have a lot of tests that we want to run with multiple encryption methods, you can make the test class abstract, put the encryption method as an abstract property, and make a derived class for each encryption method. We do similar thing in order to parameterize httpclient test on http versions

https://github.com/alinpahontu2912/runtime/blob/6732696fe982903e47edb7a49647fab0aa526920/src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs#L323-L340